pyro/distributions/permutation_flow.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.distributions.transforms import Transform
+from torch.distributions import constraints
+
+from pyro.distributions.util import copy_docs_from
+
+
+@copy_docs_from(Transform)
+class PermutationFlow(Transform):
+    """
+    A normalizing flow that reorders the input dimensions, that is, multiplies the input by a permutation matrix.
+    This is useful in between IAF transforms to increase the flexibility of the resulting distribution and
+    stabilize learning. Whilst not being an autoregressive flow, the log absolute determinate of the Jacobian is
+    easily calculable as 0. Note that reordering the input dimension between two layers of IAF is not equivalent
+    to reordering the dimension inside the MADE networks that those IAFs use; using a PermutationFlow results in a
+    distribution with more flexibility.
+
+    Example usage:
+
+    >>> from pyro.nn import AutoRegressiveNN
```

Comment:
Can you please add this class to docs/source/distributions.rst, then `make docs` and ensure docs are generated correctly? I believe you'll need to indent this code block for docs to be generated correctly.

Reply:
`make docs` seems to work now with the new class in `distributions.rst`